### PR TITLE
Add Close Button to Alert Component

### PR DESCRIPTION
**Pull request by sreidthomas**
This fix added a close button to the alert component in the design system.

### CODE REVIEW

**Comment by sreidthomas** [comment]
### **Current Issues As of April 29, 2024**

Cannot get close button to position to the top right corner

Screenshot 2024-04-29 at 10 07 53 AM


Conflicts in Alert.js file

**Comment by sreidthomas** [comment]
Am I moving in the right direction with the stories file @maxatdetroit?

File: src/stories/alert.stories.js
Comment (maxatdetroit) [inline src/stories/alert.stories.js]: In terms of code structure, this is 100% moving in the right direction: we create a storybook arg that then controls the presence of an attribute on the custom element. 

My only nit is I think we should name the arg and attribute `closeable` instead of `is-open`, since the attribute should control when there is a close button on the alert, not whether the alert is open or not.

Comment (sreidthomas) [reply src/stories/alert.stories.js]: That's a perfectly fine nit. I was honestly not sure how to name I just didn't want to bug you lol

**Comment by sreidthomas** [comment]
### **UPDATES AS OF MAY 3, 2O24:**

Somehow got the "x" to be on the component in the top right corner and it functions but not sure if its using the cod-button and the attribute is being added but its not removing or adding the "x" based on the true or false closable boolean. 

Screenshot 2024-05-03 at 7 19 17 PM

Screenshot 2024-05-03 at 7 21 14 PM

### **Also have some merge conflicts that need fixing.**

**Comment by sreidthomas** [comment]
> ### **UPDATES AS OF MAY 3, 2O24:**
> Somehow got the "x" to be on the component in the top right corner and it functions but not sure if its using the cod-button and the attribute is being added but its not removing or adding the "x" based on the true or false closable boolean.
> 
> Screenshot 2024-05-03 at 7 19 17 PM Screenshot 2024-05-03 at 7 21 14 PM
> ### **Also have some merge conflicts that need fixing.**



> ### **UPDATES AS OF MAY 3, 2O24:**
> Somehow got the "x" to be on the component in the top right corner and it functions but not sure if its using the cod-button and the attribute is being added but its not removing or adding the "x" based on the true or false closable boolean.
> 
> Screenshot 2024-05-03 at 7 19 17 PM Screenshot 2024-05-03 at 7 21 14 PM
> ### **Also have some merge conflicts that need fixing.**

### **UPDATE MAY 6, 2024:**

Close button is responding to closable boolean. 

Still not sure if it is using cod-button ---> I believe it is because of these 2 lines:
`  const closeButton = document.createElement('cod-button');`
`closeButton.className = 'btn-close';`

**Comment by sreidthomas** [comment]
### **Things to do May 6, 2024:**

- [x] Fix conflicting files issue

**Comment by sreidthomas** [comment]
The close button is positioned correctly and working. I had some merge conflicts which I attempted to fix but I am not 100% sure I did it correctly but it says All Checks Have Passed. Just need some feedback on this solution when you get time @maxatdetroit

**Comment by maxatdetroit** [comment]
@sreidthomas

> Still not sure if it is using cod-button 

You can confirm this by using dev tools inspector and inspecting the close button and seeing if it has the `cod-button` tag or not. 

I'll take a look this afternoon as well.

File: src/components/atoms/Alert/Alert.js
Comment (maxatdetroit) [inline src/components/atoms/Alert/Alert.js]: This is from an incorrect conflict resolution from dev merge. This code shouldn't be changed in this PR. It was correct on the dev branch.

File: src/stories/alert.stories.js
Comment (maxatdetroit) [inline src/stories/alert.stories.js]: This is also from an incorrect conflict resolution from dev merge. Please revert these changes.

**Review by maxatdetroit — CHANGES_REQUESTED** [review]
The code looks exactly right, but we have some changes in this PR that shouldn't be here due to incorrect dev merge conflict resolution. We should be good to merge this PR once those are reverted.

File: src/stories/alert.stories.js
Comment (sreidthomas) [reply src/stories/alert.stories.js]: I was able to revert Alert.js but alert.stories.js is not letting me do anything @maxatdetroit

Comment (sreidthomas) [reply src/stories/alert.stories.js]: This is the code that was already in this file @maxatdetroit

Comment (sreidthomas) [reply src/stories/alert.stories.js]: Screenshot 2024-05-06 at 2 03 07 PM

Comment (sreidthomas) [reply src/stories/alert.stories.js]: I copied the code when I was re-building the sandbox trying to understand. This is what was already there before I tried to fix the merge conflict

Comment (maxatdetroit) [reply src/stories/alert.stories.js]: > This is what was already there before I tried to fix the merge conflict

That's right, because the branch had old code that wasn't up-to-date with `dev`. Then, when we merged in `dev`, it tried to update this code to what is on dev (i.e. the newer code). We should have taken that newer code instead of keeping the old code.

Here's the new code from dev: https://is.gd/rXc37Q

That's what this section should look like (with the addition of your new storybook arg you added in this PR).

File: src/components/atoms/Alert/Alert.js
Comment (maxatdetroit) [reply src/components/atoms/Alert/Alert.js]: I marked this as unresolved again because this section of code still needs to be fixed to match dev. Like this: https://is.gd/2aYdxS

File: src/components/atoms/Alert/Alert.js
Comment (maxatdetroit) [inline src/components/atoms/Alert/Alert.js]: I think the revert that you mentioned was a revert to the wrong commit. Take a look at this file on dev to see what these sections should look like: https://github.com/CityOfDetroit/COD-Design-System/blob/dev/src/components/atoms/Alert/Alert.js

**Comment by sreidthomas** [comment]
Can I just copy the old code and add mines back in? I reverted back to the DEV merge and it keeps coming back like that because I did it wrong completely @maxatdetroit

**Review by maxatdetroit — APPROVED** [review]
LGTM! Thanks for fixing those merge issues.

**Comment by sreidthomas** [comment]
@jedgar1mx Its ready to merge

**Review by jedgar1mx — APPROVED** [review]

**Pull request merged**
